editor/src/components/assets.ts

```diff
@@ -42,11 +42,18 @@ export function getAllProjectAssetFiles(
   return allProjectAssets
 }
 
-function getSHA1Checksum(contents: string): string {
+function getSHA1Checksum(contents: string | Buffer): string {
   return new sha1().update(contents).digest('hex')
 }
 
-export function getProjectContentsChecksums(tree: ProjectContentTreeRoot): GithubChecksums {
+export function inferGitBlobChecksum(buf: Buffer): string {
+  return getSHA1Checksum(Buffer.concat([Buffer.from(`blob ${buf.byteLength}\0`), buf]))
+}
+
+export function getProjectContentsChecksums(
+  tree: ProjectContentTreeRoot,
+  assetChecksums: GithubChecksums,
+): GithubChecksums {
```

Comment:
does this function handle each file type we have? or is it OK that some files won't be included in `checksums`